ansible_base/resource_registry/apps.py

```diff
@@ -67,18 +67,31 @@ def initialize_resources(sender, **kwargs):
             r_type.save()
 
 
+def proxies_of_model(cls):
+    """Return models that are a proxy of cls"""
+    for sub_cls in cls.__subclasses__():
+        if sub_cls._meta.concrete_model is cls:
+            yield sub_cls
+
+
 def connect_resource_signals(sender, **kwargs):
     from ansible_base.resource_registry.signals import handlers
 
-    signals.post_save.connect(handlers.update_resource)
-    signals.post_delete.connect(handlers.remove_resource)
+    for model in handlers.get_resource_models():
+        for cls in [model] + list(proxies_of_model(model)):
```

Comment:
IMO This looks cleaner

```suggestion
        for cls in [model, *proxies_of_model(model)]:
```

Reply:
pushed in commit.